lib/fog/vsphere/requests/compute/vm_clone.rb

```diff
@@ -811,8 +814,7 @@ def modify_template_volumes_specs(vm_mob_ref, volumes)
                          device: template_volume }
             end
           end
-          new_volumes.map { |volume| volume.unit_number = volumes.index(volume) < 7 ? volumes.index(volume) : volumes.index(volume) + 1 }
```

Comment:
This is solved by copying the identifiers from the template volume?

Reply:
Yeah exactly, the problem was the volume numbers were duplicated, but the above line is fixing it.